src/main/scala/esmeta/analyzer/domain/value/ValueTypeDomain.scala

```diff
@@ -323,12 +327,7 @@ trait ValueTypeDomainDecl { self: Self =>
         Elem(StrT(names))
 
       /** type check */
-      def typeCheck(tname: String, st: AbsState): Elem =
-        val names = instanceNameSet(elem.ty)
-        if (names.isEmpty) Bot
-        else if (names == Set(tname)) Elem(TrueT)
-        else if (!names.contains(tname)) Elem(FalseT)
-        else boolTop
+      def typeCheck(tname: String, st: AbsState): Elem = boolTop
```

Comment:
121 more errors are occurred during `tycheck` due to this patch (in `dev branch` 319 → 440).

Below AOs are added after running `sbt test` (TypeCheckSmallTest)
```
"ApplyStringOrNumericBinaryOperator",
"AsyncFromSyncIteratorContinuation",
"CanBeHeldWeakly",
"CompareTypedArrayElements",
"INTRINSICS.BigInt",
"INTRINSICS.Function.prototype.bind",
"INTRINSICS.Map.groupBy",
"INTRINSICS.Object.groupBy",
"INTRINSICS.String.prototype.replace",
"IsLessThan",
"IsLooselyEqual",
"IsStrictlyEqual",
"ModuleNamespaceExoticObject.Get",
"PerformEval",
"RunJobs:clo0",
"SameValue",
"SameValueNonNumber",
"SameValueZero",
"StringGetOwnProperty",
"ThisBigIntValue",
"ThisBooleanValue",
"ThisNumberValue",
"ThisStringValue",
"ToNumber",
"ToNumeric",
"ToString",
```
`GroupBy`, `Map.groupBy` and `Object.groupBy` AOs will be good entrypoint to debug typecheck because we discussed these briefly today.

Reply:
I think large amount of errors are occurred due to the change the result of `(? x: "CompletionRecord")`. It was always evaluated to `FalseT` before fix.

Reply:
I will fix this part when refactoring the type analysis! Thanks!